### PR TITLE
Toggle full name display of users

### LINKED ISSUE #5222: [FR] Owner: Show User's full name instead of username
### Please verify that this feature request has NOT been suggested before.

- [X] I checked and didn't find a similar feature request

### Problem statement

Any Owner field tied to a user currently shows the username when printed in the GUI.

1. It's not always easily discernable who "PikachuLover5000" is
2. In conjunction with PO's, using "Attn: PikachuLover5000" in a report isn't very desirable either

### Suggested solution

By changing what value is returned based on Owner type, any "accidentally" bad usernames are filtered out of the GUI part, and only exposed to the "offending" user and admins.

I propose a very simple solution:
Return different model fields based on `owner_type`.

`owner.name` for Groups
`owner.full_name` for Users

### Describe alternatives you've considered

Pretty much the above

### Examples of other systems

_No response_

### Do you want to develop this?

- [X] I want to develop this.

### PR DESCRIPTION
**Pull request by LavissaWoW**
Would've closed #5222 if it was still open

### CODE REVIEW

File: InvenTree/users/models.py [hunk 1]
```diff
@@ -785,10 +802,17 @@ def get_api_url():  # pragma: no cover
 
     def __str__(self):
         """Defines the owner string representation."""
-        return f'{self.owner} ({self.owner_type.name})'
+        print("Setting is:", common_models.InvenTreeSetting.get_setting('DISPLAY_FULL_NAMES'))
```
Comment (SchrodingersGat) [inline InvenTree/users/models.py:805]: Please remove this debug statement :)

Comment (LavissaWoW) [reply InvenTree/users/models.py:805]: Oops :>

**Comment by SchrodingersGat** [comment]
@LavissaWoW thanks, looks good to me :)

**Comment by wolflu05** [comment]
Wouldn't it make sense to also add this new setting to PUI so that they don't get out of sync?

**Comment by SchrodingersGat** [comment]
@wolflu05 yep, that's a good point. @LavissaWoW would you be able to implement a similar change into the new react frontend?

**Comment by matmair** [comment]
In this case, adding the key of the setting somewhere here should be enough BTW
https://github.com/inventree/InvenTree/blob/d30ff8a291d945c13984d6998dce67b4091839c4/src/frontend/src/pages/Index/Settings/SystemSettings.tsx#L47-L62

**Pull request merged**